### PR TITLE
Setting up arm builder using the new runner

### PR DESCRIPTION
**Pull request by djpolygon**
Changed name of the amd64 packager

Added arm64 packager that will build on the newly added runner, this may need some iterations to iron out as the arm hosted github runners are still in beta.

### CODE REVIEW

File: .github/workflows/arm_deb_packager.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
+name: arm_deb_packager
+
+on:
+  push:
+    branches: [main, next]
+    paths:
+      - '**'
+    tags:
+      - 'v*.*.*'
+      - 'v*.*.*-*'
```
Comment (bobbinth) [inline .github/workflows/arm_deb_packager.yml:10]: I think we changed this slightly in the `next` branch ([here](https://github.com/0xPolygonMiden/miden-node/blob/next/.github/workflows/deb_packager.yml)).

**Review by bobbinth — COMMENTED** [review]
Thank you! I'll defer the full review to @Mirko-von-Leipzig - but I did leave one comment inline. More generally, we may need to rebase this on the `next` branch as this is what we use for development (and this is also where all the latest code is).

File: .github/workflows/arm_deb_packager.yml (hunk 1 above)
Comment (djpolygon) [reply .github/workflows/arm_deb_packager.yml]: Updated  in the arm packager

File: .github/workflows/arm_deb_packager.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      - name: adding version
+        run: |
+          NUMERIC_VERSION=$( echo ${{ env.GIT_TAG }} | sed 's/[^0-9.]//g' )
+          echo "VERSION=$NUMERIC_VERSION" >> $GITHUB_ENV
+
+      - name: cleaning repo
+        run: cargo clean
+
+      - name: Building for arm64
+        run: cargo build --release
```
Comment (Mirko-von-Leipzig) [inline .github/workflows/arm_deb_packager.yml:30]:
```suggestion
      - name: Building for arm64
        run: |
          cargo build --release --locked --features testing --bin miden-node
          cargo build --release --locked --features testing --bin miden-faucet
```

**Review by Mirko-von-Leipzig — COMMENTED** [review]
What's missing is an overall strategy to deal with the slight difference in devnet and testnet deploys. I assume we would want to utilize debian packages for both.

For testnet:

1. We click release, this means there is a formal version tag available `vx.y.z`
2. Generate the debian package for `vx.y.z` **and** upload these as assets to the github release
3. Deploy this package to the testnet instance

For devnet:

1. We push to `next` branch
2. Generate the debian package for `next` but **do not** upload them to any release
3. Deploy this package to the devnet instance

--------------------

Otherwise if this is just an initial attempt to see that packaging works (and should be separate from the deployment for now), then its just missing a way to differentiate between testnet and devnet.

Right now this looks decent for a proper release packaging i.e. for testnet only

**Comment by Mirko-von-Leipzig** [comment]
> Thank you! I'll defer the full review to @Mirko-von-Leipzig - but I did leave one comment inline. More generally, we may need to rebase this on the `next` branch as this is what we use for development (and this is also where all the latest code is).

Yes we'll need to do that; but only after these workflows have appeared on `main`. github only acknowledges workflows that are on the default branch i.e. `main` for us. Though once that has happened its somehow happy to run any updates on other branches to these workflows.

Its just the initial "registration" that has to land on `main` first.

**Review by Mirko-von-Leipzig — APPROVED** [review]

**Pull request merged**